CMakeLists.txt

```diff
@@ -0,0 +1,163 @@
+# instead depend on YCM.
+# In this case replace the following line with
+# find_package(YCM REQUIRED)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+### Options
+# Shared/Dynamic or Static library?
+option(BUILD_SHARED_LIBS "Build libraries as shared as opposed to static" ON)
+
+# Build test related commands?
+option(BUILD_TESTING "Create tests using CMake" OFF)
+if(BUILD_TESTING)
+    enable_testing()
+endif()
+
+# Enable RPATH support for installed binaries and libraries
+include(AddInstallRPATHSupport)
+add_install_rpath_support(BIN_DIRS "${CMAKE_INSTALL_FULL_BINDIR}"
+                          LIB_DIRS "${CMAKE_INSTALL_FULL_LIBDIR}"
+                          INSTALL_NAME_DIR "${CMAKE_INSTALL_FULL_LIBDIR}"
+                          USE_LINK_PATH)
+
+# Encourage user to specify a build type (e.g. Release, Debug, etc.), otherwise set it to Release.
+if(NOT CMAKE_CONFIGURATION_TYPES)
+    if(NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING
+        "Choose the type of build, recommanded options are: Debug or Release" FORCE)
+    endif()
+    set(IDYNFOR_BUILD_TYPES "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS ${IDYNFOR_BUILD_TYPES})
+endif()
+
+
+include(iDynForDependencies)
+
+### Compile- and install-related commands.
+add_subdirectory(src)
+
+# Create and install CMake configuration files for your project that are
+# necessary to for other projects to call find_package(LibTemplateCMake).
+#
+# Note that it is extremely important to use exactly the project name while
+# installing configuration files (you can use PROJECT_NAME variable to avoid
+# any possible error). This is required to allow find_package() to properly
+# look for the installed library in system path, in particular in Windows when
+# the installation is performed in the default path.
+#
+# install_basic_package_files() comes with many input parameters to customize
+# the configuration files. The parameters used in the following call provide
+# basic versions of CMake configuration files.
+# See install_basic_package_files() documentation found in ./cmake folder.
+#
+# Note that if your library depends from other libraries, you are probably
+# required to used the install_basic_package_files() DEPENDENCIES option.
+include(InstallBasicPackageFiles)
+install_basic_package_files(${PROJECT_NAME}
+                            VERSION ${${PROJECT_NAME}_VERSION}
+                            COMPATIBILITY AnyNewerVersion
+                            NO_EXPORT   # NO_EXPORT is required since we have only cmake scripts. To be changed in EXPORT ${${PROJECT_NAME}_TARGETS}
+                            VARS_PREFIX ${PROJECT_NAME}
+                            NO_CHECK_REQUIRED_COMPONENTS_MACRO
+                            DEPENDENCIES ${IDYNFOR_PUBLIC_DEPENDENCIES}
+                            PRIVATE_DEPENDENCIES ${IDYNFOR_PRIVATE_DEPENDENCIES})
```

Comment:
Where are these variables set?

Reply:
These variables are not set anywhere. Do you think I should add these lines?
```cpp
set(IDYNFOR_PUBLIC_DEPENDENCIES )
set(IDYNFOR_PRIVATE_DEPENDENCIES )
```

Reply:
No no, I think you should simply list the dependencies instead of ``${IDYNFOR_PUBLIC_DEPENDENCIES}``. Probably, at this stage we don't have any private dependency, so probably that line can be removed.

Reply:
> No no, I think you should simply list the dependencies instead of `${IDYNFOR_PUBLIC_DEPENDENCIES}`. Probably, at this stage we don't have any private dependency, so probably that line can be removed.

By "that line", I meant 
https://github.com/dic-iit/idynfor/blob/f8265f5e6df08671432a13adc4f709893120f0ba/CMakeLists.txt#L157
(cleary paying attention to the parenthesis)

Reply:
Thanks Ste! Done in https://github.com/dic-iit/idynfor/pull/6/commits/5cd18e51dd973d337e62129a0ad38144dbe4198a.